code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,114 @@
+
+  Background: Common Geofencing Subscriptions setup
+    Given the resource "/geofencing-subscriptions/v0/subscriptions"  as geofencing Url                                                            |
+    And the header "Content-Type" is set to "application/json"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
+    And the request body is set by default to a request body compliant with the schema
+
+  @geofencing_subscriptions_01_Create_geofencing_subscription_for_a_device
+  Scenario:  Create geofencing subscription
+    Given they use the geofencing url
+    When they create subscription
+    Then Response code is 201
+
+  @geofencing_subscriptions_02_Operation_to_retrieve_list_of_subscriptions
+  Scenario: Get a list of subscriptions.
+    Given they use the geofencing url
+    When they get all subscriptions
+    Then Response code is 200
+
+
+  @geofencing_subscriptions_03_Operation_to_retrieve_subscription_based_on_the_provided_ID
+  Scenario: Get a subscription based on provided id.
+    Given they use the geofencing url
+    When they get subscription for subscription-id
+    Then Response code is 200
+
+  @geofencing_subscriptions_04_Operation_to_delete_subscription_based_on_the_provided_ID
+  Scenario: Delete a subscription based on provided id.
+    Given they use the geofencing url
+    When they delete subscription for subscription-id
+    Then Response code is 204
+
+  @geofencing_subscriptions_05_Create_invalid_geofencing_subscription_for_a_device
+  Scenario:  Create geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they create subscription with invalid parameter
+    Then Response code is 400
+
+  @geofencing_subscriptions_06_Get_invalid_geofencing_subscription_for_a_device
+  Scenario:  Get geofencing subscription with invalid subscription-id which is not available
+    Given they use the geofencing url
+    When they get subscription with invalid subscription-id which is not available
+    Then Response code is 404
+
+
+  @geofencing_subscriptions_07_Delete_invalid_geofencing_subscription_for_a_device
+  Scenario:  Delete geofencing subscription with invalid parameter
+    Given they use the geofencing url
+    When they delete subscription with invalid subscription-id
+    Then Response code is 404
+
+  @geofencing_subscriptions_08_Invalid_method_geofencing_subscription_for_a_device
+  Scenario:  Update geofencing subscription
+    Given they use the geofencing url
+    When they update subscription
+    Then Response code is 405
+
+
+  @geofencing_subscriptions_09_creation_of_subscription_for_subscribed_expired_time_in_past
+  Scenario:  Subscribed expire time in past for geofencing subscription
+    Given they use the geofencing url
+    When they create subscription with expire time in past
+    Then Response code is 400
```

Comment:
could that be 422? But I am also ok with 400.

Reply:
400 will be fine i guess for this scenario